src/components/GroupBanner.tsx

```diff
@@ -27,20 +35,44 @@ export default function GroupBanner(props: {
   const [user] = useUserContext();
   const transcriptCount = props.countTranscripts ? props.group.transcripts.length : 0;
   const [isJoiningMeeting, setJoining] = useState(false);
+  const [isMeetingOccupied, setOccupancy] = useState(false);
   const launchMeeting = async (groupId: string) => {
-    setJoining(true);
-    try {
-      const link = await tClientBrowser.myGroups.generateMeetingLink.mutate({ groupId: groupId });
-      window.location.href = link;
-    } catch (e) {
-      toast.error((e as Error).message, { autoClose: false });
-    } finally {
-      // More time is needed to redirect to the meeting page. Keep it spinning.
-      // We should uncomment this line if we pop the page in a new window.
-      // setJoining(false);
+    if ((await tClientBrowser.myGroups.countOngoingMeeting.query()) !== 0) {
+      setOccupancy(true);
+    } else {
+      setJoining(true);
+      try {
+        const link = await tClientBrowser.myGroups.generateMeetingLink.mutate({ groupId: groupId });
```

Comment:
Also, we need to change the table design of `OngoingMeetingCount`, because we should allow expected participants of the ongoing meeting to join. We need to change it to:

```
class OngoingMeetings ... {
  @PrimaryKey
  TMAdminUserId: string;
  
  @PrimaryKey
  meetingLink: string;
}
```

`joinMeeting` should first see if the current, non-expired meeting link is already in this table, if yes return the link. if no, then check the row count of this table.

Can you create a new PR just for refactoring this table and `updateOngoingMeetingCounts`? Then we can come back to this PR.